pkgs/development/python-modules/tinygrad/default.nix

```diff
@@ -209,7 +213,10 @@ buildPythonPackage rec {
     changelog = "https://github.com/tinygrad/tinygrad/releases/tag/v${version}";
     license = lib.licenses.mit;
     maintainers = with lib.maintainers; [ GaetanLepage ];
-    # Requires unpackaged pyobjc-framework-libdispatch and pyobjc-framework-metal
-    broken = stdenv.hostPlatform.isDarwin;
+    badPlatforms = [
```

Comment:
I don't understand what the advantage of badPlatforms over broken is here.

Reply:
Yeah, the TLDR is that `broken` implies the package is broken because of a version or feature. It shows an error message only saying its broken. However, `badPlatforms` will throw up a message saying the package doesn't work on the platform. I've got #348192 which will improve things so `badPlatforms` will work optimally.

Reply:
>  broken implies the package is broken because of a version or feature.

Isn't that exactly the case? Darwin is a supported platform, but we're running into an obscure segfault with the version update

Reply:
Somewhat, the [manual](https://nixos.org/manual/nixpkgs/stable/#var-meta-badPlatforms) is more specific.

> The list of Nix [platform types](https://github.com/NixOS/nixpkgs/blob/b03ac42b0734da3e7be9bf8d94433a5195734b19/lib/meta.nix#L75-L81) on which the package is known not to be buildable.

Reply:
We can continue this discussion in https://github.com/NixOS/nixpkgs/pull/348192/files#r1858923693